torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -173,6 +222,11 @@ void DistEngine::computeDependencies(
     for (const auto& recvBackwardEdge : recvBackwardEdges) {
       graphTask->exec_info_[recvBackwardEdge.function.get()].needed_ = true;
     }
+    // Mark all output nodes as needed.
+    for (const auto& outputEdge : outputEdges) {
```

Comment:
Not sure why we need to explicitly mark those outputEdge functions. Would those functions already be marked properly in "graphTask->init_to_execute" in line 219?

Reply:
Deleted according to @albanD 's suggestion.